### PR TITLE
Header file for Matrix object and Matrix operations

### PR DESCRIPTION
**Pull request by aniketsharma00411**

### CODE REVIEW

File: examples/get_mat.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    // Getting a vector of rows
+    std::vector<double> row_2 = sliced_mat.get_row(2);
+    for (int i = 0; i < row_2.size(); i++)
+        std::cout << row_2[i] << "\t";
+    std::cout << std::endl;
+
+    // Getting a vector of columns
+    std::vector<double> col_3 = sliced_mat.get_col(3);
+    for (int i = 0; i < col_3.size(); i++)
```
Comment (ashok-arora) [inline examples/get_mat.cpp:30]: Is there a `view` or `print` function to replace this for loop?

Comment (aniketsharma00411) [reply examples/get_mat.cpp:30]: This for loop is for printing a vector<double>.

 vector<double> will only used for making graphs.

File: include/matrix.hpp [hunk 2]
```diff
@@ -0,0 +1,480 @@
+#ifndef _matrix_hpp_
+#define _matrix_hpp_
+
+#include <all.h>
+
+class Matrix {
+  private:
+    std::vector<std::vector<double>> dob_mat;
```
Comment (ashok-arora) [inline include/matrix.hpp:8]: It would be better to replace `dob_mat` with `double_mat`

File: include/matrix.hpp [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+        mat.to_double();
+    }
+
+    return mat;
+}
+
+/* Method to return the Tranpose of a Matrix
+   The method will return a Matrix object whose dimensions will be (col_length(), row_length())
+*/
+Matrix Matrix::T() {
```
Comment (ashok-arora) [inline include/matrix.hpp:143]: Transpose would be a better name for the function since we are making it similar to numpy

**Comment by ashok-arora** [comment]
This merge request has been moved to https://github.com/ashok-arora/Matrix

**Pull request closed without merging**